functorch/_src/vmap.py

```diff
@@ -40,8 +41,20 @@ def register_torch_return_types():
 
 register_torch_return_types()
 
-# Checks that all args-to-be-batched have the same batch dim size
 
+# Temporary OrderedDict registration as pytree
+def _odict_flatten(d):
+    return list(d.values()), list(d.keys())
+
+
+def _odict_unflatten(values, context):
+    return OrderedDict((key, value) for key, value in zip(context, values))
+
+
+_register_pytree_node(OrderedDict, _odict_flatten, _odict_unflatten)
```

Comment:
Can you add a really quick test somewhere so we don't accidentally regress on this? Something that tests (1) calling flatten returns the values, (2) calling flatten than unflatten returns an equivalent odict should be sufficient.